plugin/trino-hive/src/main/java/io/trino/plugin/hive/PartitionsSystemTableProvider.java

```diff
@@ -68,11 +80,20 @@ public PartitionsSystemTableProvider(HivePartitionManager partitionManager)
         }
 
         SchemaTableName sourceTableName = PARTITIONS.getSourceTableName(tableName);
-        HiveTableHandle sourceTableHandle = metadata.getTableHandle(session, sourceTableName);
-
-        if (sourceTableHandle == null) {
+        Table sourceTable = metadata.getMetastore()
+                .getTable(new HiveIdentity(session), sourceTableName.getSchemaName(), sourceTableName.getTableName())
+                .orElse(null);
+        if (sourceTable == null || isDeltaLakeTable(sourceTable) || isIcebergTable(sourceTable)) {
             return Optional.empty();
         }
+        verifyOnline(sourceTableName, Optional.empty(), getProtectMode(sourceTable), sourceTable.getParameters());
+        HiveTableHandle sourceTableHandle = new HiveTableHandle(
```

Comment:
NOTE that compared to `HiveMetadata#getTableHandle` I didn't include here anymore the check

```
        // we must not allow system tables due to how permissions are checked in SystemTableAwareAccessControl
        if (getSourceTableNameFromSystemTable(systemTableProviders, tableName).isPresent()) {
            throw new TrinoException(HIVE_INVALID_METADATA, "Unexpected table present in Hive metastore: " + tableName);
        }
```